algebra/src/fft/domain/mixed_radix_2_domain.rs

```diff
@@ -0,0 +1,409 @@
+use crate::{FpParameters, PrimeField};
+use crate::{multicore::Worker, EvaluationDomainImpl};
+use rayon::prelude::*;
+use std::fmt;
+
+/// Defines a domain over which finite field (I)FFTs can be performed. Works
+/// only for fields that have a large multiplicative subgroup of size that is
+/// a power-of-2 and a small multiplicative subgroup of size that is a power of
+/// q, with q being a prime number. The evaluation domain will have size 2^k * q^s.
+/// In this case, k steps of radix-2 algorithm are applied at the beginning of
+/// the algorithm, while the rest of the transform is carried out by s steps of the
+/// radix-q algorithm.
```

Comment:
actually, the code below does it in the reverse order: first the p-adic coset FFTs, and then the duadic.